moveit_core/trajectory_processing/test/test_cartesian_speed.cpp

```diff
@@ -0,0 +1,184 @@
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Benjamin Scholz, based off test_time_parameterization.cpp by Ken Anderson */
+
+#include <gtest/gtest.h>
+#include <fstream>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/robot_state/cartesian_interpolator.h>
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_trajectory/robot_trajectory.h>
+#include <moveit/trajectory_processing/iterative_time_parameterization.h>
+#include <moveit/trajectory_processing/cartesian_speed.h>
+#include <moveit/utils/robot_model_test_utils.h>
+//#include <moveit/planning_scene_interface/planning_scene_interface.h>
+//#include <moveit/move_group_interface/move_group_interface.h>
+
+// Static variables used in all tests
+moveit::core::RobotModelConstPtr RMODEL = moveit::core::loadTestingRobotModel("panda");
+robot_trajectory::RobotTrajectory TRAJECTORY(RMODEL, "panda_arm");
+
+// Vector of distances between waypoints
+std::vector<double> WAYPOINT_DISTANCES;
+
+// Name of logger
+const char* LOGGER_NAME = "trajectory_processing";
+
+// Initialize one-joint, straight-line trajectory
+bool initStraightTrajectory(robot_trajectory::RobotTrajectory& trajectory)
+{
+  const robot_model::JointModelGroup* group = trajectory.getGroup();
+  if (!group)
+  {
+    ROS_ERROR_NAMED(LOGGER_NAME, "Need to set the group");
+    return false;
+  }
+  // Get state of the robot
+  moveit::core::RobotState state(trajectory.getRobotModel());
+
+  trajectory.clear();
+  // Initial waypoint
+  state.setVariablePosition("panda_joint1", 0);
+  state.setVariablePosition("panda_joint2", -0.785);
+  state.setVariablePosition("panda_joint3", 0);
+  state.setVariablePosition("panda_joint4", -2.356);
+  state.setVariablePosition("panda_joint5", 0);
+  state.setVariablePosition("panda_joint6", 1.571);
+  state.setVariablePosition("panda_joint7", 0.785);
+  trajectory.addSuffixWayPoint(state, 0.0);
+  // First waypoint (+0.3 m in X direction)
+  state.setVariablePosition("panda_joint1", 0.00011058924053135735);
+  state.setVariablePosition("panda_joint2", 0.15980591412916012);
+  state.setVariablePosition("panda_joint3", -0.000269206763021151);
+  state.setVariablePosition("panda_joint4", -1.4550637907602342);
+  state.setVariablePosition("panda_joint5", 0.0006907805230834268);
+  state.setVariablePosition("panda_joint6", 1.61442119426705);
+  state.setVariablePosition("panda_joint7", 0.7845267455355481);
+  trajectory.addSuffixWayPoint(state, 0.0);
+  WAYPOINT_DISTANCES.push_back(0.3);
+
+  // Second waypoint (+0.3 m in Y direction)
```

Comment:
Are these waypoints *incremental* or all w.r.t. the initial waypoint?
I think it might be useful to give the actual x/y/z values in the base frame here as well if the model ever changes....

Reply:
Imo this is fine as is.